static/src/javascripts/projects/commercial/modules/messenger/background.ts

```diff
@@ -0,0 +1,219 @@
+	Object.assign(background.style, specStyles);
+};
+
+const setCtaURL = (
+	ctaURL: string,
+	backgroundParent: HTMLElement,
+): HTMLElement => {
+	const ctaURLAnchor = document.createElement('a');
+	ctaURLAnchor.href = ctaURL;
+	ctaURLAnchor.target = '_new';
+	ctaURLAnchor.appendChild(backgroundParent);
+	ctaURLAnchor.style.width = '100%';
+	ctaURLAnchor.style.height = '100%';
+	return ctaURLAnchor;
+};
+
+const renderBottomLine = (
+	background: HTMLElement,
+	backgroundParent: HTMLElement,
+): void => {
+	background.style.position = 'fixed';
+	const bottomLine = document.createElement('div');
+	bottomLine.classList.add('ad-slot__line');
+	bottomLine.style.position = 'absolute';
+	bottomLine.style.width = '100%';
+	bottomLine.style.bottom = '0';
+	bottomLine.style.borderBottom = '1px solid #dcdcdc';
+	backgroundParent.appendChild(bottomLine);
+};
+
+const setupParallax = (
+	adSlot: HTMLElement,
+	background: HTMLElement,
+	backgroundParent: HTMLElement,
+) => {
+	background.style.position = 'absolute';
+	adSlot.style.position = 'relative';
+
+	const onScroll = (background: HTMLElement) =>
+		fastdom
+			.measure(() => background.getBoundingClientRect())
+			.then((rect) =>
+				fastdom.mutate(() => {
+					const backgroundHeight = rect.height;
+					const windowHeight = window.innerHeight;
+
+					// we should scroll at a rate such that we don't run out of background (when non-repeating)
+					const parallaxBackgroundMovement = Math.floor(
+						(rect.bottom / (windowHeight + backgroundHeight)) * 130,
+					);
+
+					background.style.backgroundPositionY = `${parallaxBackgroundMovement}%`;
+				}),
+			);
+
+	const onIntersect: IntersectionObserverCallback = (entries) =>
+		entries
+			.filter((entry) => entry.isIntersecting)
+			.forEach(() => {
+				window.addEventListener(
+					'scroll',
+					() => void onScroll(background),
+					{
+						passive: true,
+					},
+				);
+				void onScroll(background);
+			});
+
+	const observer = new IntersectionObserver(onIntersect, {
+		rootMargin: '10px',
+	});
+
+	observer.observe(backgroundParent);
+};
+
+const setupBackground = async (
+	specs: BackgroundSpecs,
+	adSlot: HTMLElement,
+): Promise<void> => {
+	const { backgroundParent, background } = createParent(specs.scrollType);
+
+	return fastdom.mutate(() => {
+		setBackgroundStyles(specs, background);
+
+		if (specs.scrollType === 'parallax') {
+			setupParallax(adSlot, background, backgroundParent);
+		}
+
+		// fixed background is very similar to interscroller, generally with a smaller height
+		if (specs.scrollType === 'fixed') {
+			adSlot.style.position = 'relative';
+			if (isDCR) {
+				background.style.position = 'fixed';
+			}
+
+			if (specs.backgroundColor) {
+				backgroundParent.style.backgroundColor = specs.backgroundColor;
+			}
+		}
+
+		if (specs.scrollType === 'interscroller') {
+			if (isDCR) {
+				adSlot.style.height = '85vh';
+				adSlot.style.marginBottom = '12px';
+				adSlot.style.position = 'relative';
+			}
```

Comment:
Very minor so not super important but I can see a couple of options here:
- promote the CSS to DCR's `ArticleContainer` therefore removing the `isDCR` check
- inject the styling for both Frontend and DCR so removing the check and CSS for Frontend

I think I _slightly_ prefer injection for both so that all the logic and styling for interscrollers is self contained.